app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,42 @@
+class PostsController < ApplicationController
+    before_action :authenticate_user!
+
+    def create
+        @post = current_user.posts.build(post_params)
+        if @post.save
+            redirect_back(fallback_location: root_path)
+            flash[:notice] = "Post successfully created"
+        else
+            flash[:error] = @post.errors.full_messages
+            redirect_back(fallback_location: root_path)
```

Comment:
redirect is a full request, when it fail consider using render, no need for another request and pass your error with `flash.now` 
[OPTIONAL] - but the best way to do it. less requests is good.

Reply:
Can you show us how to use it, without a new action, because we don't have one.